core-bundle/src/Cron/PurgeOptInTokensCron.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Cron;
+
+use Contao\CoreBundle\Monolog\ContaoContext;
+use Contao\CoreBundle\OptIn\OptIn;
+use Contao\CoreBundle\ServiceAnnotation\CronJob;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @CronJob("hourly")
+ */
+class PurgeOptInTokensCron
+{
+    /**
+     * @var OptIn
+     */
+    private $optIn;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    public function __construct(OptIn $optIn, LoggerInterface $logger = null)
+    {
+        $this->optIn = $optIn;
+        $this->logger = $logger;
+    }
+
+    public function __invoke(): void
+    {
+        $this->optIn->purgeTokens();
+
+        if (null !== $this->logger) {
+            $this->logger->info('Purged the expired double opt-in tokens', ['contao' => new ContaoContext(__METHOD__, ContaoContext::CRON)]);
```

Comment:
this should only be logged if something was purged, otherwise it pollutes the back end log every hour for no reason (e.g. even if a system does not have registration at all)

Reply:
`OptIn::purgeTokens` currently only returns `void`. Can we change the return type of a public function in a bugfix release? Although I could also query the database before and after in order to figure out if and how many opt in tokens were purged.

Reply:
I think we can, since we're only adding stuff. Anyone using the current implementation would ignore the returned value.

Reply:
That won't be possible, since it's part of the interface. So I am counting the entries before and after now.